### ISSUE TITLE
Add `clearable` behavior to date and time picker inputs

### ISSUE THREAD

**Issue by nwhittaker** [Esri/calcite-design-system#9440]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

The _number_, _text_, and _time zone_ inputs all implement a `clearable` prop that allows the user to empty the input's value. This issue requests extending the same functionality to the _date_ and _time_ components.

### Acceptance Criteria

* A `clearable` attribute/prop is present on the `input-date-picker` and `input-time-picker` components.
* The clear button is visible when the input contains a non-empty value -- even if the input is required.
* Clicking/activating the button clears the input's value. If the input is required, its `valueMissing` validity state becomes enabled.
* The inputs' `messageOverrides` prop contains a `clear` entry to override the clear button's label.
* See issue #11802 for designs for this

### Which Component

`input-date-picker` and `input-time-picker`

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Field Apps

**monday.com sync:** #9365319803

**Comment by geospatialem** [comment]
Added `design` for expertise consultation across the components.

**Comment by ashetland** [comment]
Related to #6640 for clear button styling consistency across the system. Both `input-date-picker` and `input-time-picker` are already on the list 😎 in that issue.

**Comment by ashetland** [comment]
Noted above that design specs can be found in #11802. 

![Image](https://github.com/user-attachments/assets/7c451302-73b7-42a6-abd7-37da9555ce60)

**Comment by ashetland** [comment]
@Esri/calcite-pes, this _might_ be blocked by #8624. Input Date Picker `range` currently uses two inputs stacked together which would necessitate adding clear buttons to both inputs. The design specs call for a single clear button (or [esc] key press) to clear the entire set.

<img width="1876" height="623" alt="Image" src="https://github.com/user-attachments/assets/47ecd19f-c07f-47cf-9b39-2d13fec51562" />

**Comment by geospatialem** [comment]
Blocked by https://github.com/Esri/calcite-design-system/issues/8624, as mentioned by [Aaron above](https://github.com/Esri/calcite-design-system/issues/9440#issuecomment-3473843094). We should try to accommodate for 26.R1 if the blocking issue can land so we can proceed with the request. 

cc @eriklharper @ashetland @Esri/calcite-pes

**Comment by eriklharper** [comment]
> [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes), this _might_ be blocked by [#8624](https://github.com/Esri/calcite-design-system/issues/8624). Input Date Picker `range` currently uses two inputs stacked together which would necessitate adding clear buttons to both inputs. The design specs call for a single clear button (or [esc] key press) to clear the entire set.

Is the use case of each range input having its own clear button being considered?  Functionally easy to achieve either, just curious if there are users asking for that specifically or if the only use case that is desired is to clear both values in the range.

**Comment by ashetland** [comment]
Unblocking this. Input masking (#8624) shouldn't affect the ability of the clear button on Input Date Picker to clear all values.

**Comment by eriklharper** [comment]
Upon reviewing Abraham's PR, I realized the design spec doesn't ask the question if the clear button should appear for partial input values like this where the value hasn't been committed yet, but there is content in some of the fields?

<img width="192" height="43" alt="Image" src="https://github.com/user-attachments/assets/73076b0b-54bb-4399-9276-c00815b3194a" />

**Comment by ashetland** [comment]
> Upon reviewing Abraham's PR, I realized the design spec doesn't ask the question if the clear button should appear for partial input values like this where the value hasn't been committed yet, but there is content in some of the fields?
> 
> <img alt="Image" width="192" height="43" src="https://private-user-images.githubusercontent.com/821864/620263748-73076b0b-54bb-4399-9276-c00815b3194a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODM3MTk3MjksIm5iZiI6MTc4MzcxOTQyOSwicGF0aCI6Ii84MjE4NjQvNjIwMjYzNzQ4LTczMDc2YjBiLTU0YmItNDM5OS05Mjc2LWMwMDgxNWIzMTk0YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNzEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDcxMFQyMTM3MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZjkwNTJiMTU1NmExOWFiNDU1ZDNlN2E1NzFjMDMyMjQ5ZmVjNmYyN2MxMDQ0ZWEwZjY3ZjM5OTgwN2M3OGU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.2CjxeLZa5rWqazHna9k1Sq8BSCWSFBITVYjS0wgHKfw">

Great question. A tricky one for sure. I think the current UX of showing the clear button when the value is stored makes sense to me, but I'm curious what others think. @Esri/calcite-designers

**Comment by eriklharper** [comment]
Regarding my earlier comment about clearing uncommitted values, I think this is fine UX and probably don't need to extend it to this use case.  We can look at the `clearable` feature as giving the user the ability to clear the committed value.  Tested that out locally and its looking good!

**Comment by matgalla** [comment]
Agree with it being an indicator of a stored value that can be cleared.

**Comment by ashetland** [comment]
@aPreciado88 I just noticed this small correction to make on the vertical range divider. There should be padding on the left.

<img width="434" height="232" alt="Image" src="https://github.com/user-attachments/assets/009dff56-ea48-4f3c-8311-c2fa2e4687f2" />

[See v5.1](https://codepen.io/ashetland/pen/BypPLxz) for correct padding.

**Comment by isaacbraun** [comment]
Verified on `5.2.0-next.29` via [CodePen](https://codepen.io/editor/isaacbraun/pen/019fab44-9264-7ae4-829e-a7d4da091538) that the above acceptance criteria were met, and:
- The added action tokens style the `clearable` and/or toggle actions as described by their descriptions.

While verifying, found that the `validity` property does not update immediately. When submitting a form, the values are correct, but when logging them after `change` events they are one tick/update behind. Opened a new issue to address: #14904
